app/models/pregnancy.rb

```diff
@@ -33,6 +33,15 @@ class Pregnancy
   # Validations
   validates :created_by,
             presence: true
+  before_update :check_other_validations?
+
+  def check_other_validations?
+    dcaf_soft_pledge.present? &&
+    Patient.first.name.present? &&
+    Patient.first.appointment_date.present? &&
+    Patient.first.clinic.name.present?
```

Comment:
Swap the `clinic.name` here out for `clinic_name` -- we haven't implemented Clinic objects yet, which is something I should have done a better job of documenting! Sorry, that's on me.